katsdpcontroller/product_config.py

```diff
@@ -374,6 +414,62 @@ def from_config(cls,
         )
 
 
+class NgcAntennaChannelisedVoltageStream(AntennaChannelisedVoltageStreamBase):
+    """Real antenna-channelised-voltage stream (next-gen correlator).
+
+    It currently only supports wideband.
+    """
+
+    stream_type: ClassVar[str] = 'ngc.antenna_channelised_voltage'
+    _valid_src_types: ClassVar[_ValidTypes] = {'sim.dig.raw_antenna_voltage'}
+
+    def __init__(self, name: str, src_streams: Sequence[Stream], *,
+                 n_chans: int) -> None:
+        if n_chans < 1 or (n_chans & (n_chans - 1)) != 0:
```

Comment:
Nice trick! I'm going to add that to my mental toolbox to check for a power of 2.